File: legacy/src/main/rs/emulate/legacy/config/floor/DefaultFloorDefinition.java
package rs.emulate.legacy.config.floor;

import rs.emulate.legacy.config.DefaultConfigDefinition;
import rs.emulate.legacy.config.SerializableProperty;

import java.util.HashMap;
import java.util.Map;

import static rs.emulate.legacy.config.Properties.alwaysFalse;
import static rs.emulate.legacy.config.Properties.asciiString;
import static rs.emulate.legacy.config.Properties.unsignedByte;
import static rs.emulate.legacy.config.Properties.unsignedTribyte;
import static rs.emulate.legacy.config.floor.FloorProperty.*;

/**
 * A {@link DefaultConfigDefinition} for {@link FloorDefinition}s.
 *
 * @param <T> The type of {@link FloorDefinition} this default is for.
 * @author Major
 */
public class DefaultFloorDefinition<T extends FloorDefinition> extends DefaultConfigDefinition<T> {

	@Override
	protected Map<Integer, SerializableProperty<?>> init() {
		Map<Integer, SerializableProperty<?>> properties = new HashMap<>();

		properties.put(1, unsignedTribyte(COLOUR, 0));
		properties.put(2, unsignedByte(TEXTURE, 0));
		properties.put(5, alwaysFalse(SHADOWED, true));
		properties.put(6, asciiString(NAME, null));
		properties.put(7, unsignedTribyte(MINIMAP_COLOUR, 0));

		return properties;
	}

}

File: legacy/src/main/kotlin/rs/emulate/legacy/config/floor/DefaultFloorDefinition.kt
package rs.emulate.legacy.config.floor

import rs.emulate.legacy.config.DefaultConfigDefinition
import rs.emulate.legacy.config.Properties.alwaysFalse
import rs.emulate.legacy.config.Properties.asciiString
import rs.emulate.legacy.config.Properties.unsignedByte
import rs.emulate.legacy.config.Properties.unsignedTribyte
import rs.emulate.legacy.config.SerializableProperty
import rs.emulate.legacy.config.floor.FloorProperty.*
import java.util.HashMap

/**
 * A [DefaultConfigDefinition] for [FloorDefinition]s.
 *
 * @param T The type of [FloorDefinition] this default is for.
 */
class DefaultFloorDefinition<T : FloorDefinition> : DefaultConfigDefinition<T>() {

    override fun init(): Map<Int, SerializableProperty<*>> {
        val properties = HashMap<Int, SerializableProperty<*>>()

        properties[1] = unsignedTribyte(Colour, 0)
        properties[2] = unsignedByte(Texture, 0)
        properties[5] = alwaysFalse(Shadowed, true)
        properties[6] = asciiString(Name, "null") // TODO defaults to null (the value) in client
        properties[7] = unsignedTribyte(MinimapColour, 0)

        return properties
    }

}
